src/shogun/clustering/KMeans.cpp

```diff
@@ -77,35 +77,30 @@ void CKMeans::set_initial_centers(SGMatrix<float64_t> centers)
 
 void CKMeans::set_random_centers(SGVector<float64_t> weights_set, SGVector<int32_t> ClList, int32_t XSize)
 {
+	/* initialise the cluster centroids randomly among all data */
 	CDenseFeatures<float64_t>* lhs=
 		CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
+	REQUIRE(lhs!=NULL, "lhs Pointer should not be NULL");
```

Comment:
This error message does not help anyone.
BUT I just realised the cast will always work as CEuclideandistance is type safe
You can make it an assert, or just remove it